packages/frontend/core/src/modules/workspace/properties/adapter.ts

```diff
@@ -0,0 +1,161 @@
+ *
+ * Question for enhancement in the future:
+ * May abstract the adapter for each property type, e.g. PagePropertiesAdapter, SchemaAdapter, etc.
+ * So that the adapter could be more focused and easier to maintain (like assigning default values)
+ * However the properties for an abstraction may not be limited to a single yjs map.
+ */
+export class WorkspacePropertiesAdapter {
+  // provides a easy-to-use interface for workspace properties
+  private readonly proxy: WorkspaceAffineProperties;
+  public readonly properties: Y.Map<any>;
+
+  constructor(private readonly workspace: Workspace) {
+    // check if properties exists, if not, create one
+    const rootDoc = workspace.doc;
+    this.properties = rootDoc.getMap(AFFINE_PROPERTIES_ID);
+    this.proxy = createYProxy(this.properties);
+
+    // fixme: deal with migration issue?
+    this.ensureRootProperties();
+  }
+
+  private ensureRootProperties() {
+    // todo: deal with schema change issue
+    // fixme: may not to be called every time
+    defaultsDeep(this.proxy, {
+      schema: {
+        pageProperties: {
+          custom: {},
+          system: {
+            journal: {
+              id: PageSystemPropertyId.Journal,
+              name: 'Journal',
+              source: 'system',
+              type: PagePropertyType.Date,
+            },
+            tags: {
+              id: PageSystemPropertyId.Tags,
+              name: 'Tags',
+              source: 'system',
+              type: PagePropertyType.Tags,
+              options: this.workspace.meta.properties.tags?.options ?? [], // better use a one time migration
+            },
+          },
+        },
+      },
+      favouriates: {},
+      pageProperties: {},
+    });
+  }
+
+  private ensurePageProperties(pageId: string) {
+    // fixme: may not to be called every time
+    defaultsDeep(this.proxy.pageProperties, {
+      [pageId]: {
+        custom: {},
+        system: {
+          [PageSystemPropertyId.Journal]: {
+            id: PageSystemPropertyId.Journal,
+            value: false,
+          },
+          [PageSystemPropertyId.Tags]: {
+            id: PageSystemPropertyId.Tags,
+            value: [],
+          },
+        },
+      },
+    });
+  }
+
+  get schema() {
+    return this.proxy.schema;
+  }
+
+  get favouriates() {
+    return this.proxy.favouriates;
+  }
+
+  get pageProperties() {
+    return this.proxy.pageProperties;
+  }
+
+  // ====== utilities ======
+
+  getPageProperties(pageId: string) {
+    return this.pageProperties[pageId];
+  }
+
+  isFavouriated(id: string, type: WorkspaceFavoriteItem['type']) {
+    return this.favouriates[id]?.type === type;
+  }
+
+  isJournalPage(id: string) {
+    this.ensurePageProperties(id);
+    return !!this.getJournalPageDate(id);
+  }
+
+  getJournalPageDate(id: string) {
+    this.ensurePageProperties(id);
+    return this.pageProperties[id].system[PageSystemPropertyId.Journal].value;
+  }
+
+  setJournalPageDate(id: string, date: string) {
```

Comment:
~~Maybe handle journal date here? Force validate and format date. I think we can do it later, but the date format should be determined as soon as possible, and should not be easily changed.~~

Maybe it's better to handle in `Journal Module` later.

Reply:
I haven't decided how to make more fine-graind abstraction for this yet. It could follow some approach of that in blocksuite, but maybe it is a bit early at this stage